src/controllers/ExperimentViewController.ts

```diff
@@ -113,6 +131,43 @@ export class ExperimentViewController {
     }
   }
 
+  /**
+   * React to any state changes
+   */
+  private eventListener(): () => void {
```

Comment:
I'm not sure this should be a method. I think a good rule of thumb is if the code isn't called from more than one place, it shouldn't be extracted. Extracting things into methods/functions _by itself_ doesn't make for better code. Try to avoid doing it if the only reason is "organizing code".